protocol/v2/ssv/spectest/msg_processing_type.go

```diff
@@ -45,52 +49,104 @@ func RunMsgProcessing(t *testing.T, test *MsgProcessingSpecTest) {
 	test.RunAsPartOfMultiTest(t, logger)
 }
 
-func (test *MsgProcessingSpecTest) runPreTesting(logger *zap.Logger) (*validator.Validator, error) {
+func (test *MsgProcessingSpecTest) runPreTesting(ctx context.Context, logger *zap.Logger) (*validator.Validator, *validator.Committee, error) {
 	var share *spectypes.Share
+	ketSetMap := make(map[phase0.ValidatorIndex]*spectestingutils.TestKeySet)
 	if len(test.Runner.GetBaseRunner().Share) == 0 {
 		panic("No share in base runner for tests")
 	}
 	for _, validatorShare := range test.Runner.GetBaseRunner().Share {
 		share = validatorShare
 		break
 	}
-	v := protocoltesting.BaseValidator(logger, spectestingutils.KeySetForShare(share))
-	v.DutyRunners[test.Runner.GetBaseRunner().RunnerRoleType] = test.Runner
-	v.Network = test.Runner.GetNetwork()
 
-	var lastErr error
-	if !test.DontStartDuty {
-		lastErr = v.StartDuty(logger, test.Duty)
+	for valIdx, validatorShare := range test.Runner.GetBaseRunner().Share {
+		ketSetMap[valIdx] = spectestingutils.KeySetForShare(validatorShare)
 	}
-	for _, msg := range test.Messages {
-		dmsg, err := queue.DecodeSignedSSVMessage(msg)
-		if err != nil {
-			lastErr = err
-			continue
+
+	var v *validator.Validator
+	var c *validator.Committee
+	var lastErr error
+	switch test.Runner.(type) {
+	case *runner.CommitteeRunner:
+		c = baseCommitteeWithRunnerSample(ctx, logger, ketSetMap, test.Runner.(*runner.CommitteeRunner))
+
+		if !test.DontStartDuty {
+			lastErr = c.StartDuty(logger, test.Duty.(*spectypes.CommitteeDuty))
+		} else {
+			c.Runners[test.Duty.DutySlot()] = test.Runner.(*runner.CommitteeRunner)
+		}
```

Comment:
its more readable to turn it around. 
reversing (`!`) `DontStartDuty` makes it hard to understand the code. 

```suggestion
		if test.DontStartDuty {
					c.Runners[test.Duty.DutySlot()] = test.Runner.(*runner.CommitteeRunner)
		} else {
		lastErr = c.StartDuty(logger, test.Duty.(*spectypes.CommitteeDuty))
		}
```